ERCS/erc-7208.md

```diff
@@ -0,0 +1,598 @@
+
+A case-by-case limited analysis is provided in the appendix.
+
+
+## Specification
+
+### Terms
+
+**ODC**: A uniquely identifiable non-fungible token. An ODC MAY store information within Properties.
+
+**Property**: A modifiable information unit stored within an ODC. Properties SHOULD be capable of undergoing modifications and MUST be able to act as an indexed data container.
+
+**Restriction**: A configuration stored within the **ODC**, that SHOULD describe under which conditions a certain **Property Manager** is allowed to modify the information stored within a certain **Property**.
+
+**Property Manager**: A type of Smart Contract that MUST implement a **PM interface** in order to manage data stored within the **ODC**.
+
+**Category**: **Property Managers** MUST be grouped in Categories that SHOULD represent access to **Properties**. Each **Property Manager** MAY be part of one or more **Categories**. The assignment of categories SHOULD be managed by Governance.
+
+### The DataStorage Library (example - not part of the ERC)
+This library implementation allows the creation of On-chain Data Containers that can store various data types, handle versions of data, and efficiently manage stored data. The `DataStorageLib` provides a system for handling data that is both efficient and flexible. The `struct DataStorageInternal` includes mappings that enable the storage of different data types.
+These are:
+ * `keyValueData` for `bytes32` key-value pairs,
+ * `keyBytesData` for storing `bytes`,
+ * `keySetData` for sets of `bytes32` values,
+ * `keyMapData` for mappings of `bytes32 => bytes32` values.
+
+Dynamic Data Versions:
+The library handles versioning of data. The `DataStorage` struct contains a 'current' version and a mapping that links versions to specific indexes in the storage. This allows the smart contract to maintain a historical record of state changes, as well as revert to previous versions if necessary.
+
+Clearing and Relocation:
+Several functions, such as `clear()`, `wipe()`, and `moveData()` are dedicated to clearing and relocating stored data. This functionality allows efficient management of stored data.
+
+Addition and Removal of Data:
+The library includes functions to set and get values of different data types. Functions such as `setValue()` and `getBytes32Value()` facilitate this functionality. The addition or removal of data is reflected in the respective set (e.g., `kvKeys`, `kbKeys`, `ksKeys`, `kmKeys`) to ensure that the library correctly keeps track of all existing keys.
+
+Efficient Data Retrieval:
+There are several getter functions to facilitate data retrieval from these storage structures. These include getting all keys (`getAllKeys()`), checking if a set contains a value (`getSetContainsValue()`), and getting all entries from a mapping (`getMapAllEntries()`).
+
+Data Deletion:
+The library provides efficient ways to delete data, like `deleteAllFromEnumerableSet()` and `deleteAllFromEnumerableMap()`.
```

Comment:
```suggestion
```

If this text isn't part of the specification, it shouldn't be part of the specification section. I'd suggest moving it to the reference implementation section?